### ISSUE TITLE
A way to wrap a multiple separate items into one dropdown

### ISSUE THREAD

**Issue by Reinmar** [ckeditor/ckeditor5-ui#322]
As described in https://github.com/ckeditor/ckeditor5-heading/issues/74#issuecomment-305224952.

The goal is to allow wrapping multiple items into one dropdown just by the toolbar configuration (or implementing a super simple plugin).

**Comment by oleq** [comment]
Do you think what we did in the Font Size/Family/Highlight is enough? Should we come up with a generic mechanism to configure dropdowns using the toolbar config (it seems like a very tricky thing, though).

**Comment by Reinmar** [comment]
It's a different kind of feature. We have APIs to wrap things in dropdowns. But it requires building the editor. We need something declarative, which could be done by anyone when integrating an editor.

**Comment by oleq** [comment]
Declarative like toolbar API/config?

**Comment by Reinmar** [comment]
Yes ;) Like in the linked comment which started this topic: https://github.com/ckeditor/ckeditor5-heading/issues/74#issuecomment-305224952.

**Comment by oleq** [comment]
I'm not sure how could we wrap a couple of items in a drop-down and assign an icon to it, though. We could specify an icon as a string somewhere (in the cfg) but what about importing it? We need an implicit `import` declaration to obtain this kind of resource; we don't have a resource manager we could use for that purpose in v5.

**Comment by Reinmar** [comment]
An SVG's source can be specified through a config file. Or we'll add [support for SVG source paths](https://github.com/ckeditor/ckeditor5-ui/issues/293) and it'll be even easier.

Besides, you don't need an icon for a normal dropdown. For a dropdown based on a split button we can use icons of buttons that we're wrapping, so that's not a problem either.

**Comment by long-lazuli** [comment]
I did a plugin especially to do this this week-end.
We plan to release it open-source as soon as we find the time to put (at least some) documentation on it.

**Comment by oleq** [comment]
I'm glad it works for you, @long-lazuli. 

But I'm afraid this kind of approach is a no-go when developers use any default [editor build](https://docs.ckeditor.com/ckeditor5/latest/builds/guides/overview.html) and they have no way to import any internals (classes, helpers) from the build. In other words: I think we should avoid any solution that involves importing/executing code at the `Editor.create()` end.

I think we could either:
* have this feature out-of-the-box like we [discussed](https://github.com/ckeditor/ckeditor5-heading/issues/74#issuecomment-305224952),
* make the toolbar population process open (e.g. event–driven) so loading a `DropdownCreator` plugin would allow it to contribute to the process – the plugin would return a dropdown when a certain syntax is found in the toolbar config
